keps/sig-api-machinery/1693-warnings/README.md

```diff
@@ -109,38 +109,39 @@ In client-go:
    3. log warnings with klog.Warn
 3. Add a process-wide warning handler (defaulting to the klog.Warn handler)
 4. Add a per-client warning handler (defaulting to the process-wide warning handler)
-
+ 
 In kubectl, configure the per-process handler to:
 
 1. dedupe warnings (only print a given warning once per invocation)
 2. log to stderr with a `Warning:` prefix
 3. color the `Warning:` prefix if stderr is a terminal and `$TERM != "dumb"` and `$NO_COLOR` is unset
-
+ 
 In kube-apiserver and kube-controller-manager, configure the process-wide handler to ignore warnings
 
 ## Design Details
-
+ 
 Server-side:
 
 * Add a handler chain filter that attaches a WarningRecorder implementation to the request context
-* Add a WarningRecord implementation that deduplicates the warning message per request and writes a `Warning` header
+* Add a WarningRecorder implementation that de-duplicates the warning message per request and writes a `Warning` header
   * the header structure is defined in [RFC2616#14.46](https://www.w3.org/Protocols/rfc2616/rfc2616-sec14.html#sec14.46)
   * warnings would be written with code `299` and warn-agent of `-`
 * In the endpoint installer, decorate handlers for deprecated APIs:
   * add the warning header
-  * increment the counter for the deprecated use metric
-  * add an audit annotation indicating the request was to a deprecated API
-
+  * increment the counter metric for the deprecated use
+  * add an audit annotation indicating the request was made to a deprecated API
```

Comment:
when you add admission capability to set a warning, I bet you want to be able to include an attribution (which actor marked it as warning).

Reply:
this annotation is independent of the warning header, and is specific to deprecated API use. admission plugins can already contribute audit annotations.

Reply:
> this annotation is independent of the warning header, and is specific to deprecated API use. admission plugins can already [contribute audit annotations which are attributed to the originating webhook](https://github.com/kubernetes/api/blob/release-1.16/admission/v1/types.go#L137-L142)

ok